core/controllers/profile.py

```diff
@@ -382,10 +382,13 @@ def get(self):
             temp_file, mode='w', compression=zipfile.ZIP_DEFLATED) as zfile:
             zfile.writestr('oppia_takeout_data.json', json.dumps(user_data))
             for image in user_images:
-                b64_png_no_header = image.b64_image_data.split(',')[1]
-                b64_png_no_header = python_utils.url_unquote_plus(
-                    b64_png_no_header)
-                b64_png_no_header = re.sub(r'\s', b'+', b64_png_no_header)
+                # Remove the 'image/png' prefixing the base64 string, replace
+                # url-encoded characters, and replace whitespace with the +
```

Comment:
Put quotes around the plus sign.

Please also add more explanation about _why_ each step needs to be done, not just what is being done. This is important for future reference. Thanks!